File: mobile-app/lib/services/auth_service.dart
import 'api_service.dart';

class AuthService {
  final ApiService _api = ApiService();

  Future<Map<String, dynamic>> login(String email, String password) async {
    return await _api.post('/auth/login', {'email': email, 'password': password});
  }

  Future<Map<String, dynamic>> register(String email, String fullName, String password) async {
    return await _api.post('/auth/register', {'email': email, 'full_name': fullName, 'password': password});
  }
}

File: web-frontend/src/services/authService.js
import api from './api';

export const authService = {
  login: async (email, password) => {
    const response = await api.post('/auth/login', { email, password });
    return response.data;
  },
  register: async (email, full_name, password) => {
    const response = await api.post('/auth/register', { email, full_name, password });
    return response.data;
  },
  googleLogin: async (idToken) => {
    const response = await api.post('/auth/google', { id_token: idToken });
    return response.data;
  },
  refreshToken: async (refreshToken) => {
    const response = await api.post('/auth/refresh', { refresh_token: refreshToken });
    return response.data;
  },
};
